### ISSUE TITLE
How to include images

### ISSUE THREAD

**Issue by machinekoder** [alexanderjulo/wiki#69]
How can I used images in my wiki? The Markdown syntax does not seem to work.

**Comment by alexanderjulo** [comment]
Can you give me some details on what exactly you tried and did not work? 🙂 Theoretically this should just work as usually!

**Comment by machinekoder** [comment]
I tried following:
```markdown
![chip](./chip_pinouts.jpg)
![chip](./chip-motor/chip_pinouts.jpg)
![chip](./chip_pinouts)
<img src="./chip_pinouts">chip<img>
```

The Markdown file is called `chip-motor.md` and the image is in `./chip-motor/chip_pinouts.jpg`

Are there any restrictions on the image format?

**Comment by alexanderjulo** [comment]
Yeah looking at the code I don't think we support static files at all at the moment. There are multiple issues at hand here and I need to think about how we can solve them best.

I see the following approaches:
(1) Expose all raw files via URL: Does not work because that might also expose the `config.py` including the secret key and potentially other files
(2) Making a directory configurable that will allow serving of static files: This would work but comes with the problem that the user will not be able to store images with the respective documents, which is kind of sad.

I am open to input before starting to implement anything. 🙂

**Comment by machinekoder** [comment]
How about exposing all files with an image extension? `png, jpg, gif, svg, ...`

**Comment by alexanderjulo** [comment]
Well the problem there is, that the list is never going to be comprehensive (i.e. you might want to expose a pdf file or even a python script but still not the config.py or things along these lines. So we will either exclude files people want to share or will include files people do not want to share, which sucks.

I'll think about it.

**Comment by machinekoder** [comment]
Or maybe both:
Whitelist and blacklist in the config.py with reasonable defaults.

**Comment by hugohonorem** [comment]
why not serve the images in a route with send_from_directory?
